portal/account/account.lib.php

```diff
@@ -0,0 +1,269 @@
+<?php
+session_start();
+if ($_SESSION['patient_portal_onsite_two'] && $_SESSION['pid']) {
+    $ignoreAuth_onsite_portal_two = true;
+}
+
+require_once("../../interface/globals.php");
+include_once("$srcdir/patient.inc");
+require_once(dirname(__FILE__) . "/../lib/portal_mail.inc");
+require_once("$srcdir/pnotes.inc");
+
+$action = isset($_REQUEST['action']) ? $_REQUEST['action'] : '';
+
+if ($action == 'set_lang') {
+    $_SESSION['language_choice'] = (int) $_REQUEST['value'];
+    echo 'okay';
+    exit();
+}
+
+if ($action == 'get_newpid') {
+    $rtn = is_new($_REQUEST['dob'], $_REQUEST['last'], $_REQUEST['first']);
+    if ((int) $rtn != 0) {
+        echo xlt("This account already exists.") . "\r\n\r\n" . xlt("If you are having troubles logging into your account.") . "\r\n" . xlt("Please contact your provider.") . "\r\n" .
+             xla("Reference this Account Id: ") . $rtn;
+        exit();
+    }
+    $rtn = attr(get_newPid());
+    echo "$rtn";
+
+    exit();
+}
+
+if ($action == 'is_new') {
+    $rtn = is_new($_REQUEST['dob'], $_REQUEST['last'], $_REQUEST['first']);
+    echo "$rtn";
+
+    exit();
+}
+
+if ($action == 'do_signup') {
+    $rtn = do_credentials($_REQUEST['pid']);
+    echo "$rtn";
+
+    exit();
+}
+
+if ($action == 'new_insurance') {
+    $pid = $_REQUEST['pid'];
+    save_insurance($pid);
+
+    exit();
+}
+
+if ($action == 'notify_admin') {
+    $pid = $_REQUEST['pid'];
+    $provider = $_REQUEST['provider'];
+    $rtn = notify_admin($pid, $provider);
+    echo "$rtn";
+
+    exit();
+}
+
+if ($action == 'cleanup') {
+    unset($_SESSION['patient_portal_onsite_two']);
+    unset($_SESSION['authUser']);
+    unset($_SESSION['pid']);
+    unset($_SESSION['site_id']);
+    unset($_SESSION['register']);
+    echo 'gone';
+    session_destroy(); // I know, makes little sense.
+}
```

Comment:
Would a switch/case make sense above (i rarely use this switch/case, but above seems like a good fit).

Reply:
I don't care for them much but use them some. Maybe an elseif chain. I always start out these libraries thinking I will turn them into a class but my libs generally are specific and not needing to be class. I don't know, here speed is not at issue so either way.